### ISSUE TITLE
Add a linkchecker for the front end

### ISSUE THREAD

**Issue by Julian** [bowtie-json-schema/bowtie#963]
We should link check every link contained in the frontend UI with an automated link checker in order to catch broken links.

**Comment by JeelRajodiya** [comment]
We can implement this using a GitHub action, 
The same way we are currently doing in webiste, see [this PR](https://github.com/json-schema-org/website/pull/297)

I like to work on this issue

**Comment by officeneerajsaini** [comment]
@JeelRajodiya, yeah, that's a great idea. Also, we need to add Vercel to our repository so that anyone making changes can submit a pull request. This way, we can check their pull request's UI with Vercel's development environment before merging it into the main branch.

**Comment by Julian** [comment]
[`lychee` ](https://github.com/lycheeverse/lychee) -- which is the actual tool being used there, seems fine/nice -- but I don't really like the flow in that PR for the JSON Schema website -- specifically it's quite odd to have to spin up the website to do the link check.

I would hope there's some more "integrated" way to run lychee over the output of vite and/or React applications, so that it's simply a final step in the build process.

If you or anyone thinks they're able to can you investigate doing things that way?

**Comment by JeelRajodiya** [comment]
> [`lychee` ](https://github.com/lycheeverse/lychee) -- which is the actual tool being used there, seems fine/nice -- but I don't really like the flow in that PR for the JSON Schema website -- specifically it's quite odd to have to spin up the website to do the link check.
> 
> I would hope there's some more "integrated" way to run lychee over the output of vite and/or React applications, so that it's simply a final step in the build process.
> 
> If you or anyone thinks they're able to can you investigate doing things that way?

Sure! I'll investigate on that

**Comment by JeelRajodiya** [comment]
I figured that our frontend is a single page react app. So when we build the project our index.html looks something like this

![image](https://github.com/bowtie-json-schema/bowtie/assets/63534268/1664e6e8-31f0-4461-9025-dd6363c2d221)

## Structure of the build folder 
![image](https://github.com/bowtie-json-schema/bowtie/assets/63534268/9e1ac1c3-964f-47b7-ad5f-629cb32872c8)

The hyperlinks are available into the main html file when the javascript file is loaded and executed using the browser. 

Link checker tools that are available on the internet checks for broken links inside a html or any other static files.
if we want to check for the broken links we'll have to run the build in a virtual browser ( that will hydrate the index.html ) and then we can check for broken links from that index.html file using tools like **lychee**. 

If we don't want to run a virtual browser, since the index.js file contains the links what we can do is write a script to check for broken links from the index.js file. 
 ## This index.js file 
![image](https://github.com/bowtie-json-schema/bowtie/assets/63534268/086134a0-3db4-4a7b-84b9-016a30f3f2dd)

## Example of links saved in the index.js file 
![image](https://github.com/bowtie-json-schema/bowtie/assets/63534268/bf4c097b-64f5-4532-9740-e08f40f066ff)

**Comment by JeelRajodiya** [comment]
I found a npm package [puppeteer](https://www.npmjs.com/package/puppeteer?activeTab=readme) that can crawl a SPA (Single-Page Application) and generate pre-rendered content (i.e. "SSR" (Server-Side Rendering)).

**Comment by JeelRajodiya** [comment]
@Julian let me know if I should proceed or not

**Comment by Julian** [comment]
Thanks for the research -- I'm not fully thrilled with that kind of setup, but I don't yet have any other suggestion. Let's hold off for a bit then and I'll have to have a look at trying to find something that illustrates, or else I'll give up and we can go with something like that, but I want to look first.
